functions/src/scripts/add-auto-resolution.ts

```diff
@@ -0,0 +1,55 @@
+// Existing contracts don't have auto resolutions (Time + resolution). Let's add them.
+
+import * as admin from 'firebase-admin'
+
+import { initAdmin } from './script-init'
+initAdmin()
+
+import { getValues } from '../utils'
+import { Contract } from '../../../common/contract'
+import { DAY_MS } from '../../../common/util/time'
+import { batchedWaitAll } from '../../../common/util/promise'
+
+const firestore = admin.firestore()
+
+async function addAutoResolutionToContracts() {
+  console.log('Adding auto resolution to existing contracts')
+
+  const contracts = await getValues<Contract>(
+    firestore.collection('contracts').where('isResolved', '==', false)
+  )
+
+  console.log('Loaded', contracts.length, 'contracts')
+
+  await batchedWaitAll(
+    contracts.map((c) => async () => {
+      addAutoResolutionToContract(c)
```

Comment:
Need an await here. Or you could return. Best is probably `contracts.map((c) => () => addAutoResolutionToContract(c))` which returns for you.